sdk/servicebus/azure-servicebus/tests/servicebus_preparer.py

```diff
@@ -0,0 +1,378 @@
+import hashlib
+from collections import namedtuple
+
+from azure.mgmt.servicebus import ServiceBusManagementClient
+from azure.mgmt.servicebus.models import SBQueue, AccessRights
+
+from azure_devtools.scenario_tests.exceptions import AzureTestError
+
+from devtools_testutils import (
+    ResourceGroupPreparer, AzureMgmtPreparer, FakeResource
+)
+
+from devtools_testutils.resource_testcase import RESOURCE_GROUP_PARAM
+
+SERVICEBUS_DEFAULT_AUTH_RULE_NAME = 'RootManageSharedAccessKey'
+SERVICEBUS_NAMESPACE_PARAM = 'servicebus_namespace'
+SERVICEBUS_TOPIC_PARAM = 'servicebus_topic'
+SERVICEBUS_SUBSCRIPTION_PARAM = 'servicebus_subscription'
+SERVICEBUS_QUEUE_PARAM = 'servicebus_queue'
+SERVICEBUS_AUTHORIZATION_RULE_PARAM = 'servicebus_authorization_rule'
+SERVICEBUS_QUEUE_AUTHORIZATION_RULE_PARAM = 'servicebus_queue_authorization_rule'
+
+# Service Bus Namespace Preparer and its shorthand decorator
+class ServiceBusNamespacePreparer(AzureMgmtPreparer):
+    def __init__(self,
+                 name_prefix='',
+                 sku='Standard', location='westus',
+                 parameter_name=SERVICEBUS_NAMESPACE_PARAM,
+                 resource_group_parameter_name=RESOURCE_GROUP_PARAM,
+                 disable_recording=True, playback_fake_resource=None,
+                 client_kwargs=None):
+        super(ServiceBusNamespacePreparer, self).__init__(name_prefix, 24,
+                                                     disable_recording=disable_recording,
+                                                     playback_fake_resource=playback_fake_resource,
+                                                     client_kwargs=client_kwargs)
+        self.location = location
+        self.sku = sku
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.parameter_name = parameter_name
+        self.connection_string = ''
+
+    def create_resource(self, name, **kwargs):
+        if self.is_live:
+            self.client = self.create_mgmt_client(ServiceBusManagementClient)
+            group = self._get_resource_group(**kwargs)
+            namespace_async_operation = self.client.namespaces.create_or_update(
+                group.name,
+                name,
+                {
+                    'sku': {'name': self.sku},
+                    'location': self.location,
+                }
+            )
+            self.resource = namespace_async_operation.result()
+
+            key = self.client.namespaces.list_keys(group.name, name, SERVICEBUS_DEFAULT_AUTH_RULE_NAME)
+            self.connection_string = key.primary_connection_string
+            self.key_name = key.key_name
+            self.primary_key = key.primary_key
+        else:
+            self.resource = FakeResource(name=name, id=name)
+            self.connection_string = 'https://microsoft.com'
```

Comment:
I don't think code will ever run as these tests always run live - however I think returning a connection string like this would cause an parsing error in the clients. Maybe we should just raise here if the test isn't live? Or do we still have some tests that use this?

Reply:
Ah my fast-and-loose-placeholder comes back to bite me :P  I don't believe we do, I put this in place largely for continuity with the other preparers.  (I additionally know we had a pipe dream to, in the future, have some sort of non-live-test contingent)  If you're not opposed I'm inclined to put a stub in that's actually well formed.

Reply:
Coming up with "fake" test details is a real pain.... You have to fake them well enough to actually hit the error scenarios you want to test - which always takes me multiple iterations... >_<